daemon/util/Thread.h

```diff
@@ -73,15 +74,15 @@ class Thread
 
 private:
 	static Mutex m_threadMutex;
-	static int m_threadCount;
+	static std::atomic_uint m_threadCount;
 #ifdef WIN32
 	HANDLE m_threadObj = 0;
 #else
 	pthread_t m_threadObj = 0;
 #endif
-	bool m_running = false;
-	bool m_stopped = false;
-	bool m_autoDestroy = false;
+	std::atomic_bool m_running{false};
+	std::atomic_bool m_stopped{false};
+	std::atomic_bool m_autoDestroy{false};
```

Comment:
`m_autoDestroy` is set before the thread is started and is never changed later; no need to protect.

As for `m_running` and `m_stopped` - is protecting here really necessary? The thread worker functions usually often check thread status via `IsStopped` to decide if the processing must be aborted. For example in article downloader it is checked after every (small) chunk of downloaded data - that maybe thousands times per second (in every download thread).

I thought the synchronisation can be omitted here since we work with a primitive type (bool) which is read/written with a single CPU operation (hopefully, yeah). Worst case scenario - the thread which reads the value may get the older value of the variable but that's not critical as it will eventually get the correct value somewhat later. Is this asking for troubles or a legitimate optimisation?

Reply:
Ok, I will revert `m_autoDestroy` to bool. About the other two, in my opinion it's better to program safe than try to optimize something that doesn't really make a difference in performance, but it's your call if you want this to be reverted as well.